pkg/cleanup/cleanup.go

```diff
@@ -0,0 +1,52 @@
+package cleanup
+
+import (
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Cleaner is an interface for the cleanup methods.
+type Cleaner interface {
+	CleanShadowServices() error
+}
+
+// Ensure the Prepare fits the Preparer interface
+var _ Cleaner = (*Cleanup)(nil)
+
+// Cleanup holds the clients for the various resource controllers.
+type Cleanup struct {
+	client k8s.Client
+	log    logrus.FieldLogger
+}
+
+// NewCleanup returns an initialized cleanup object.
+func NewCleanup(log logrus.FieldLogger, client k8s.Client) Cleaner {
+	return &Cleanup{
+		client: client,
+		log:    log,
+	}
+}
+
+// CleanShadowServices deletes all shadow services from the cluster.
+func (c *Cleanup) CleanShadowServices() error {
+	serviceList, err := c.client.GetKubernetesClient().CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{
+		LabelSelector: "app=maesh",
+	})
+	if len(serviceList.Items) == 0 {
```

Comment:
Can't `serviceList` be nil?

Reply:
I don't think so, its not a pointer:

https://godoc.org/k8s.io/api/core/v1#ServiceList

although I guess an empty slice is nil isn't it? but wouldn't that also still work with the range?

Reply:
I have removed this check since we just iterate over the items after the error check.